### PR TITLE
fix: send dest info for failed events

### PR DESCRIPTION
**Pull request by sanpj2292**
## Description of the change

- We need to send `destination` information to fill up the tags of `workspaceId` and `destinationId` for `router_transform_num_jobs` measurement for `router` transform destinations --- ✅ 
- We need to send `destination` information to fill up the tags of `workspaceId` and `destinationId` for `router_transform_num_jobs` measurement for `batch` transform destinations --- ✅ 
- We are following the same convention for `process` transformation --- 🤷🏼‍♂️ 
- For transformer proxy this information is currently not needed --- ❌ 

Without this change for atleast router transformation and batch transformation features, for successfully transformed jobs we would have workspaceId & destinationId rightly bound but if the job has failed, we would not have `workspaceId` and `destinationJob` information as it is not being sent when an error is thrown in transformer

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by saikumarrs** [comment]
Isn't the destination info already available with the server as it makes the request to the transformer?

File: src/v0/util/index.js [hunk 1]
```diff
@@ -1640,7 +1640,7 @@ const handleRtTfSingleEventError = (input, error, reqMetadata) => {
     ...getEventReqMetadata(input)
   });
 
-  return resp;
+  return { ...resp, destination: input.destination };
```
Comment (saikumarrs) [inline src/v0/util/index.js:1643]: If this is definitely needed, we might have to update all the route transform destinations that don't use this method to handle the event-level errors.

**Comment by sanpj2292** [comment]
> Isn't the destination info already available with the server as it makes the request to the transformer?

The thing is you have to send it in the response, for the measurement `router_transform_num_jobs` to make sure that it contains `destinationId` and `workspaceId`  in case of **failures**

File: src/v0/util/index.js (hunk 1 above)
Comment (sanpj2292) [reply src/v0/util/index.js:1643]: Added the information to wherever it is needed

**Review by koladilip — DISMISSED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**